### PR TITLE
Deprecate device.qubit_set  in cirq_google.

### PR DESCRIPTION
**Pull request by MichaelBroughton**
Yet more of #4744 .

It also looks like now we can get rid of the device shim @mpharrigan , @dstrain115 (will leave to seperate PR).

### CODE REVIEW

File: cirq-google/cirq_google/workflow/qubit_placement_test.py [hunk 1]
```diff
@@ -22,6 +22,12 @@
 class TestDevice(cirq.Device):
     def __init__(self):
         self.qubits = cirq.GridQubit.rect(2, 8)
+        neighbs = [(a, b) for a in self.qubits for b in self.qubits if a.is_adjacent(b)]
```
Comment (95-martin-orion) [inline cirq-google/cirq_google/workflow/qubit_placement_test.py:25]: Nit: spare the two extra characters to make this read nicely?

**Review by 95-martin-orion — APPROVED** [review]

**Comment by mpharrigan** [comment]
so wait, the only way to get a device's qubits now is to go through a level of indirection?

**Comment by MichaelBroughton** [comment]
Yes, this was discussed in the design review for devices p2 and we signed off on moving `qubit_set` to metadata and keeping just the validation components present for core `Device`.

File: cirq-google/cirq_google/workflow/qubit_placement_test.py (hunk 1 above)
Comment (MichaelBroughton) [reply cirq-google/cirq_google/workflow/qubit_placement_test.py]: Done.

**Pull request merged**